core/src/main/java/io/substrait/expression/proto/ExpressionProtoConverter.java

```diff
@@ -465,57 +471,68 @@ public Expression visit(io.substrait.expression.Expression.Window expr) throws R
         .build();
   }
 
+  private static class LiteralToWindowBoundOffset
+      extends AbstractExpressionVisitor<Long, RuntimeException> {
+
+    @Override
+    public Long visitFallback(io.substrait.expression.Expression expr) {
+      throw new RuntimeException(
+          String.format("Expected positive integer for Window Bound offset, received: %s", expr));
+    }
+
+    private static long offsetIsPositive(long offset) {
+      if (offset >= 1) {
+        return offset;
+      }
+      throw new RuntimeException(
+          String.format("Expected positive offset for Window Bound offset, recieved: %d", offset));
+    }
+
+    @Override
+    public Long visit(io.substrait.expression.Expression.I8Literal expr) throws RuntimeException {
+      return offsetIsPositive(expr.value());
+    }
+
+    @Override
+    public Long visit(io.substrait.expression.Expression.I16Literal expr) throws RuntimeException {
+      return offsetIsPositive(expr.value());
+    }
+
+    @Override
+    public Long visit(io.substrait.expression.Expression.I32Literal expr) throws RuntimeException {
+      return offsetIsPositive(expr.value());
+    }
+
+    @Override
+    public Long visit(io.substrait.expression.Expression.I64Literal expr) throws RuntimeException {
+      return offsetIsPositive(expr.value());
+    }
+  }
+
   private Expression.WindowFunction.Bound toBound(io.substrait.expression.WindowBound windowBound) {
     var boundedKind = windowBound.boundedKind();
-    Expression.WindowFunction.Bound bound = null;
-    switch (boundedKind) {
-      case CURRENT_ROW -> bound =
-          Expression.WindowFunction.Bound.newBuilder()
-              .setCurrentRow(Expression.WindowFunction.Bound.CurrentRow.getDefaultInstance())
-              .build();
+    return switch (boundedKind) {
+      case CURRENT_ROW -> Expression.WindowFunction.Bound.newBuilder()
+          .setCurrentRow(Expression.WindowFunction.Bound.CurrentRow.getDefaultInstance())
+          .build();
       case BOUNDED -> {
         WindowBound.BoundedWindowBound boundedWindowBound =
             (WindowBound.BoundedWindowBound) windowBound;
-        var offset = boundedWindowBound.offset();
-        boolean isPreceding = boundedWindowBound.direction() == WindowBound.Direction.PRECEDING;
-        io.substrait.expression.Expression.I32Literal offsetLiteral =
-            (io.substrait.expression.Expression.I32Literal) offset;
```

Comment:
The cast of the didn't always succeed. Reading plans in from protos, the offset is encoded as a long which is converted to an I64Literal. Additionally, depending on how Calcite encodes the offset as well, these could potentially be I8 or I16. The possibilities are handled in the LiteralToWindowBoundOffset visitor.